### PR TITLE
[v6r18] Reschedule job when payload exited with status DErrno.EWMSRESC

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES
*WMS
NEW: The application error codes, when returned, are passed to the JobWrapper, and maybe interpreted.
NEW: The JobWrapperTemplate can reschedule a job if the payload exits with status DErrno.EWMSRESC
ENDRELEASENOTES

### CODE REVIEW

**Comment by atsareg** [comment]
Looks OK to me. I would just define a constant instead of a magic "111" with an understandable name

**Comment by fstagni** [comment]
OK I am changing it. Also I will add a better debug.

**Comment by fstagni** [comment]
I added several changes to allow the application to return an error code to the JobWrapper. This impacts the core Workflow modules, so please review carefully.

File: WorkloadManagementSystem/JobWrapper/JobWrapper.py [hunk 1]
```diff
@@ -438,6 +443,10 @@ def execute( self, arguments ):
         self.__report( 'Completed', 'Application Finished Successfully', sendFlag = True )
       elif not watchdog.checkError:
         self.__report( 'Completed', 'Application Finished With Errors', sendFlag = True )
+        if status in (DErrno.EWMSRESC, DErrno.EWMSRESC & 255): # the status will be truncated to 0xDE (222)
```
Comment (atsareg) [inline WorkloadManagementSystem/JobWrapper/JobWrapper.py]: I think the status could never be DErrno.EWMSRESC here, so the exact match to the second value is enough

**Pull request merged**